.github/workflows/_runner_ondemand_slurm.yaml

```diff
@@ -93,3 +94,17 @@ jobs:
             -e DOCKERD_IN_RUNNER=true \
             ghcr.io/yhtang/actions-runner-dind:ubuntu-22.04  # use personal repo for the time being to avoid auth/cost issues
           EOF
+
+          echo "SLURM_JOB_ID=$(cat ${SLURM_JOB_ID_FILE})" >> $GITHUB_OUTPUT
+
+      - name: Wait for SLURM job to complete
+        shell: bash -x -e {0}
+        run: |
+          while ssh ${{ secrets.CLUSTER_LOGIN_USER }}@${{ vars.HOSTNAME_SLURM_LOGIN }} squeue -j${{ steps.submit.outputs.SLURM_JOB_ID }} > /dev/null 2>&1; do echo "wait"; sleep 15; done
```

Comment:
Would it be more robust to exit only if the squeue command returns that the job status is completed? This looks like it'll fail if `ssh` fails to connect (for w/e networking reason)

Reply:
Makes sense. In fact we could harden all places of SSH usage for such possibilities. Maybe worth another PR for it.